### PR TITLE
FIX Run TextCollectorTask as part of push instead of pull

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-tx-translator/issues/11

After merging, create a `1.1` branch from `1` and tag `1.1.0`

### CODE REVIEW

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
It shouldn't be part of push _nor_ part of pull. It should be its own configuration option.

If it has to be push or pull, it's _more_ a part of pull (i.e. locally get all of the translation strings) than it is part of push (i.e. update the translation strings in the remote store).

**Comment by emteknetnz** [comment]
Updated

File: README.md [hunk 1]
```diff
@@ -24,24 +24,31 @@ Delete this token from github once you have completed updating translations.
...
 
 Note that the valid values for the boolean variables are 1, true, on, 0, false and off.
 
-- `TX_GITHUB_API_TOKEN` (required) - the github token with write access to create pull-requests
-- `TX_SITE` (required) - the url of a local silverstripe site to run i18nTextCollectorTask against. `http://` will be automatically added if protocol is omitted
+- `TX_GITHUB_API_TOKEN` (required) - the github token with write access to create pull-requestsi18nTextCollectorTask against. `http://` will be automatically added if protocol is omitted
 - `TX_PULL` (default `1`) - pull latest translations from transifex, run i18nTextCollectorTask and update translation files
 - `TX_PUSH` (default `0`) - push new source strings to transifex
+- `TX_COLLECT` (default `0`) - run i18nTextCollectorTask on local site
+- `TX_SITE` (required if `TX_COLLECT` is `1`) - the url of a local silverstripe site to run
```
Comment (ChloeSartorelli) [inline README.md]:
```suggestion
- `TX_GITHUB_API_TOKEN` (required) - the github token with write access to create pull-requests
- `TX_PULL` (default `1`) - pull latest translations from transifex, run i18nTextCollectorTask and update translation files
- `TX_PUSH` (default `0`) - push new source strings to transifex
- `TX_COLLECT` (default `0`) - run i18nTextCollectorTask on local site
- `TX_SITE` (required if `TX_COLLECT` is `1`) - the url of a local silverstripe site to run i18nTextCollectorTask against. `http://` will be automatically added if protocol is omitted
```

File: src/Translator.php [hunk 2]
```diff
@@ -85,10 +91,6 @@ private function checkEnv(): void
         } catch (ProcessFailedException $e) {
             throw new RuntimeException('Could not find wget command. Please install it.');
         }
-        $this->txSite = getenv('TX_SITE');
-        if (!$this->txSite) {
-            throw new RuntimeException('TX_SITE environment variable is not defined');
-        }
         if (strpos($this->txSite, 'http://') !== 0 && strpos($this->txSite, 'https://') !== 0) {
             $this->txSite = 'http://' . $this->txSite;
         }
```
Comment (ChloeSartorelli) [inline src/Translator.php]: This `txSite` validation needs to be moved to after `txSite` is assigned.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (emteknetnz) [reply README.md]: Opps. Updated

File: src/Translator.php (hunk 2 above)
Comment (emteknetnz) [reply src/Translator.php]: Updated

File: src/Translator.php [hunk 3]
```diff
@@ -85,13 +91,6 @@ private function checkEnv(): void
...
-        $this->txSite = getenv('TX_SITE');
-        if (!$this->txSite) {
-            throw new RuntimeException('TX_SITE environment variable is not defined');
-        }
-        if (strpos($this->txSite, 'http://') !== 0 && strpos($this->txSite, 'https://') !== 0) {
-            $this->txSite = 'http://' . $this->txSite;
-        }
         if (!parse_url($this->txSite, PHP_URL_HOST)) {
             throw new RuntimeException('SITE environment variable is not a valid url');
         }
```
Comment (ChloeSartorelli) [inline src/Translator.php]: Sorry, I thought I had captured this part in my previous review as well. This also needs to be moved to after the assignment - and probably only be done if `$this->doCollectStrings` is truthy.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Translator.php (hunk 3 above)
Comment (emteknetnz) [reply src/Translator.php]: Updated

**Review by GuySartorelli — APPROVED** [review]
Works well locally.

**Pull request merged**